clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2589,23 +2589,25 @@ within the cluster. This attribute is ignored on the host.
 If ``intel::use_stall_enable_clusters`` is applied to a function called from a device
 kernel, the attribute is ignored and it is not propagated to the kernel.
 
+``intel::use_stall_enable_clusters]`` is a function attribute, not a kernel one.
```

Comment:
Is this correct? Does it not apply to SYCL kernels? I understand we shouldn't propagate it but I assume it should still be applied to kernel if directly applied.

Reply:
> Is this correct? Does it not apply to SYCL kernels? 

[[intel::use_stall_enable_clusters]] is a function attribute, not a kernel one. It has to be applied to the function it is attached to. It will not be collected and applied to the kernel.


>>I understand we shouldn't propagate it but I assume it should still be applied to kernel if directly applied.
This is the problem. We had this change on https://github.com/intel/llvm/pull/2734.

Reply:
My understanding is you just need to remove the warning on L578, and move CodeGen code to `StartFunction ` (already done in this patch) to allow the attribute to be applied to all functions. 

I think you still need the code for `DirectlyCalled` on L574 to properly handle attribute applied directly on SYCL kernel. With this change, for SYCL kernels, the attribute remains attached to `operator()`. I could be wrong but this does not seem correct to me. @erichkeane could you please verify?

Reply:
Our kernel should still be calling operator(), so we can leave it attached to operator() is my understanding.  This is basically just 'not' a SYCL attribute, it is a general language attribute, it doesn't impact the kernel itself is my understanding.

Reply:
Here is the problem that i am trying to fix here:
 
When [[intel::use_stall_enable_clusters]] is applied to a function that isn't directly called from a SYCL kernel function, we get warning: 'use_stall_enable_clusters' attribute allowed only on a function directly called from a SYCL kernel function; attribute ignored [-Wignored-attributes]
 [[intel::use_stall_enable_clusters]]
^
 1 warning generated.

When this happens, the stall_enable decoration isn't applied.  This is incorrect behavior.  The whole purpose of stall_enable is to partition the code into a function and apply the attribute there.

Reply:
I understood that to mean "make this like every other function attribute instead of like the special SYCL ones" and so there's no reason for the docs to mention kernels at all or for SemaSYCL.cpp to move the attribute onto anything else.

Reply:
Ok Thanks for clarifications everyone! In that case, I think I agree with @AaronBallman's suggestion above. I recommend changing docs. It is quite confusing at the moment. 

```
When applied to a lambda or function call operator (of a function object)
on device, this requests, to the extent possible, that statically-scheduled
clusters handle stalls using a stall-enable signal to freeze computation
within the cluster. This attribute is ignored on the host.
```

This needs to be updated. Also, should it still be ignored in host code? I think the current change does not do this.

Reply:
I don't think it should be ignored in host mode as it's just a regular ol' function attribute now. So I agree that the docs need updating.

Reply:
Thank you everyone for the clarification. I will update doc and patch.